enterprise/server/util/chunker/chunker.go

```diff
@@ -0,0 +1,122 @@
+package chunker
+
+import (
+	"context"
+	"io"
+
+	"github.com/buildbuddy-io/buildbuddy/server/util/status"
+	"github.com/jotfs/fastcdc-go"
+)
+
+type WriteFunc func([]byte) error
```

Comment:
I'm sure you considered this -- but asking so I understand. WDYT about including the chunk index in this function, and then having a single chunkWriteFunction like:

type WriteFunc func(chunkIndex int, data []byte) error

It would make this library code a little simpler to handle, at the expense of maybe more complexity in the pebble cache calling code where you'll want to distinguish these two cases and need to detect if there are multiple writes or not.